docs/addons/argo-cd.md

```diff
@@ -23,8 +25,92 @@ new EksBlueprint(app, 'my-stack-name', addOns, [], {
 });
 ```
 
+The above will use `argocd` namespace and install all the components. In order to bootstrap workloads you will need to change the default ArgoCD admin password and add repositories as specified in the [Getting Started](https://argoproj.github.io/argo-cd/getting_started/#port-forwarding) documentation.
+
+## Setting Admin Password from AWS Secrets Manager
+
+```typescript
+const argoCDAddOn = new ArgoCDAddOn({
+    adminPasswordSecretName: `your-secret-name`
+});
+const addOns: Array<ClusterAddOn> = [ argoCDAddOn ];
+```
+
+The attribute `adminPasswordSecretName` is the logical name of the secret in [AWS Secret Manager](https://aws.amazon.com/secrets-manager/). Note, that when deploying to multiple regions, the secret is expected to be replicated to each region. 
+
+Inside ArgoCD, the admin password is stored as a `bcrypt` hash. This step will be performed by the framework and stored in the ArgoCD admin `secret`. 
+
+You can change the admin password through the Secrets Manager, but it will require rerunning the provisioning pipeline. Automatic sync with the Secrets Manager will be added in the future. 
+
+## Bootstrapping 
+
+The SSP framework provides an approach to bootstrap workloads and/or additional add-ons from a customer GitOps repository. In a general case, the bootstrap GitOps repository may contains an [App of Apps](https://argoproj.github.io/argo-cd/operator-manual/cluster-bootstrapping/#app-of-apps-pattern) that points to all workloads and add-ons.  
+
+In order to enable bootstrapping, the add-on allows passing an `ArgoApplicationRepository` at construction time. The following repository types are supported at present:
```

Comment:
do we want to make this more generic? Will Flux need repo support eventually?

Reply:
We could do that, but auth methods are limited to those supported by Argo atm.

Reply:
gotcha